larpandoracontent/LArHelpers/LArHierarchyHelper.cc

```diff
@@ -33,30 +35,30 @@ LArHierarchyHelper::MCHierarchy::~MCHierarchy()
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
-void LArHierarchyHelper::MCHierarchy::FillHierarchy(const MCParticleList &mcParticleList, const CaloHitList &caloHitList,
-    const bool foldToPrimaries, const bool foldToLeadingShowers)
+void LArHierarchyHelper::MCHierarchy::FillHierarchy(
+    const MCParticleList &mcParticleList, const CaloHitList &caloHitList, const bool foldToPrimaries, const bool foldToLeadingShowers)
 {
-    auto predicate = [](const MCParticle *pMC) { return std::abs(pMC->GetParticleId()) == NEUTRON; };
+    const auto predicate = [](const MCParticle *pMCParticle) { return std::abs(pMCParticle->GetParticleId()) == NEUTRON; };
     m_mcToHitsMap.clear();
     for (const CaloHit *pCaloHit : caloHitList)
     {
         try
         {
-            const MCParticle *pMC{MCParticleHelper::GetMainMCParticle(pCaloHit)};
-            m_mcToHitsMap[pMC].emplace_back(pCaloHit);
+            const MCParticle *pMCParticle{MCParticleHelper::GetMainMCParticle(pCaloHit)};
+            m_mcToHitsMap[pMCParticle].emplace_back(pCaloHit);
         }
-        catch(const StatusCodeException&)
+        catch (const StatusCodeException &)
         {
         }
     }
 
+    MCParticleSet primarySet;
+    m_pNeutrino = LArHierarchyHelper::GetMCPrimaries(mcParticleList, primarySet);
+    MCParticleList primaries(primarySet.begin(), primarySet.end());
```

Comment:
Presumably want to sort the MCParticleList, after populating it via an unordered set. Should be a useful sort function in relevant helper classes (of LArContent, maybe even SDK).